<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FT8DemodSettingsDialog</class>
 <widget class="QDialog" name="FT8DemodSettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>349</width>
    <height>333</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Liberation Sans</family>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>FT8 demod settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="decoderGroup">
     <property name="title">
      <string>Decoder</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="threadsGroup">
        <item>
         <widget class="QLabel" name="decoderNbThreadsLabel">
          <property name="text">
           <string>Number of threads</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="decoderNbThreads">
          <property name="toolTip">
           <string>Number of decoding threads </string>
          </property>
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>12</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="decoderTimeBudgetLabel">
          <property name="text">
           <string>Time budget (s)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="decoderTimeBudget">
          <property name="toolTip">
           <string>Time limit for the decoder (s)</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>0.500000000000000</double>
          </property>
          <property name="maximum">
           <double>5.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>0.500000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="osdLayout">
        <property name="spacing">
         <number>2</number>
        </property>
        <item>
         <widget class="ButtonSwitch" name="osdEnable">
          <property name="toolTip">
           <string>Toggle Ordered Statistics Decoding</string>
          </property>
          <property name="text">
           <string>OSD</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="osdDepthlLabel">
          <property name="text">
           <string>Depth</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDial" name="osdDepth">
          <property name="maximumSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Order Statistics Decoding depth</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>6</number>
          </property>
          <property name="pageStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="osdDepthText">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="osdLDPCThresholdLabel">
          <property name="text">
           <string>LDPC Thr</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDial" name="osdLDPCThreshold">
          <property name="maximumSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Number of correct LDPC bits required to activate OSD </string>
          </property>
          <property name="minimum">
           <number>60</number>
          </property>
          <property name="maximum">
           <number>83</number>
          </property>
          <property name="pageStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>70</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="osdLDPCThresholdText">
          <property name="text">
           <string>60</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="verifyOSD">
          <property name="toolTip">
           <string>Verify OSD decoded message against valid callsigns</string>
          </property>
          <property name="layoutDirection">
           <enum>Qt::RightToLeft</enum>
          </property>
          <property name="text">
           <string>Vfy</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="bandsGroup">
     <property name="title">
      <string>Band presets</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>2</number>
      </property>
      <item>
       <widget class="QTableWidget" name="bands">
        <property name="toolTip">
         <string>Predefined frequencies </string>
        </property>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderMinimumSectionSize">
         <number>15</number>
        </attribute>
        <attribute name="verticalHeaderDefaultSectionSize">
         <number>15</number>
        </attribute>
        <column>
         <property name="text">
          <string>Name</string>
         </property>
         <property name="toolTip">
          <string>Band displayed name</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>F (kHz)</string>
         </property>
         <property name="toolTip">
          <string>Base frequency (kHz)</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>off (kHz)</string>
         </property>
         <property name="toolTip">
          <string>Channel offset frequency (kHz)</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="bandsControlLayout">
        <item>
         <widget class="QPushButton" name="addBand">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Add band</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../../../sdrgui/resources/res.qrc">
            <normaloff>:/create.png</normaloff>:/create.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deleteBand">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Remove selected  band</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../../../sdrgui/resources/res.qrc">
            <normaloff>:/bin.png</normaloff>:/bin.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="moveBandUp">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Move up selected  band</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../../../sdrgui/resources/res.qrc">
            <normaloff>:/arrow_up.png</normaloff>:/arrow_up.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="moveBandDown">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Move down selected  band</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../../../sdrgui/resources/res.qrc">
            <normaloff>:/arrow_down.png</normaloff>:/arrow_down.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="restoreBandPresets">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Restore default band presets</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../../../sdrgui/resources/res.qrc">
            <normaloff>:/recycle.png</normaloff>:/recycle.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ButtonSwitch</class>
   <extends>QToolButton</extends>
   <header>gui/buttonswitch.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../sdrgui/resources/res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>FT8DemodSettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>FT8DemodSettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
